test/server/routes/user.test.js

```diff
@@ -0,0 +1,148 @@
+    chai.request(server)
+      .post(`${BASE_URL}${newBook.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(409);
+        expect(response.body.message).to.be.a('string');
+        expect(response.body.message).to.equal('book already in your favourites');
+        done();
+      });
+  });
+
+  it('should not favorite book if the book is not the database', (done) => {
+    chai.request(server)
+      .post(`${BASE_URL}${nonExistingBook.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(404);
+        expect(response.body.message).to.be.a('string');
+        expect(response.body.message).to.equal('book not found');
+        done();
+      });
+  });
+
+  it('should not favorite book if the book id is of wrong format', (done) => {
+    chai.request(server)
+      .post(`${BASE_URL}${InvalidBook.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(400);
+        expect(response.body.errors).to.be.a('object');
+        expect(response.body.errors.params.bookId).to.equal('bookId value must be at least 1 and an integer');
+        done();
+      });
+  });
+
+  it('should not favorite book if the book id is of wrong format', (done) => {
+    chai.request(server)
+      .post(`${BASE_URL}${InvalidBook1.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(400);
+        expect(response.body.errors).to.be.a('object');
+        expect(response.body.errors.params.bookId).to.equal('bookId value must be at least 1 and an integer');
+        done();
+      });
+  });
+
+  it('should not favorite book if the book id is of wrong format', (done) => {
+    chai.request(server)
+      .post(`${BASE_URL}${InvalidBook2.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(400);
+        expect(response.body.errors).to.be.a('object');
+        expect(response.body.errors.params.bookId).to.equal('bookId value must be at least 1 and an integer');
+        done();
+      });
+  });
+});
+
+
+describe('Unfavorite Book Test', () => {
+  it('should unfavorite book if the book is already a favorite and also in the database', (done) => {
+    chai.request(server)
+      .delete(`${BASE_URL}${newBook.id}`)
+      .query({ token: patronToken })
+      .end((error, response) => {
+        expect(response).to.have.status(200);
+        expect(response.body.message).to.be.a('string');
+        expect(response.body.message).to.equal('unfavorited successfully');
+        expect(response.body.book).to.be.a('object');
+        expect(response.body.book).to.have.property('authorName');
+        expect(response.body.book).to.have.property('authorName');
```

Comment:
Duplicate expect(response.body.book).to.have.property('authorName')'

Reply:
done